locopy/redshift.py

```diff
@@ -201,10 +201,12 @@ def copy(self, table_name, s3path, delim="|", copy_options=None):
         """
         if not self._is_connected():
             raise DBError("No Redshift connection object is present.")
-
-        copy_options = add_default_copy_options(copy_options)
+        if "PARQUET" not in copy_options:
+            copy_options = add_default_copy_options(copy_options)
+        if delim:
+            copy_options = [f"DELIMITER '{delim}'"] + copy_options
         copy_options_text = combine_copy_options(copy_options)
-        base_copy_string = "COPY {0} FROM '{1}' " "CREDENTIALS '{2}' " "DELIMITER '{3}' {4};"
+        base_copy_string = "COPY {0} FROM '{1}' " "CREDENTIALS '{2}' " "{4};"
         try:
             sql = base_copy_string.format(
                 table_name, s3path, self._credentials_string(), delim, copy_options_text
```

Comment:
Let's remove `delim` since it is captured in `copy_options_text`. We can also change `base_copy_string = "COPY {0} FROM '{1}' " "CREDENTIALS '{2}' " "{4};"` to `base_copy_string = "COPY {0} FROM '{1}' " "CREDENTIALS '{2}' " "{3};"`

Reply:
oh nvm I think I understand what you mean

Reply:
👍